internal/preferences.go

```diff
@@ -178,7 +186,8 @@ func (p *InMemoryPreferences) RemoveValue(key string) {
 
 // NewInMemoryPreferences creates a new preferences implementation stored in memory
 func NewInMemoryPreferences() *InMemoryPreferences {
-	p := &InMemoryPreferences{}
-	p.values = make(map[string]interface{})
-	return p
+	return &InMemoryPreferences{
+		values:    make(map[string]interface{}),
+		waitGroup: &sync.WaitGroup{},
```

Comment:
Go recommends not using a variable name that is the same as the type - can you think of any shorter alternatives?

Reply:
Would using `wg` be okay?

Reply:
Yes, I think we have done that elsewhere